skbeam/core/roi.py

```diff
@@ -757,3 +759,65 @@ def auto_find_center_rings(avg_img, sigma=1, no_rings=4, min_samples=3,
         edge_pts_xy = edge_pts_xy[~inliers]
 
     return center, image, radii
+
+def find_center_rings_hough(image, num_rings, rad_range, intensity_thresh, sigma, v_range = None):
+    """
+    Take angled cut or multiple incremented cuts of a image data.
+
+    Parameters
+    ----------
+    image : ndarray
+        Image data
+    num_rings : int
+        Number of candidate circles to be searched for
+    rad_range : array
+        The min, max, and increment for range of candidate radii to be searched
+    intensity_thresh : array
+        Min and max of intensity to be used in canny filter
+    sigma : float
+        Standard deviation applied in canny filter
+    v_range : array
+        Min and max for matplotlib colormap
+
+    Returns
+    -------
+    center_pt : array
+        Average center of the candidate circles
+    radii: array
+        radii of the candidate circles
+
+    """
+
+    low_thresh = intensity_thresh[0]
+    hi_thresh = intensity_thresh[1]
+    sigma = sigma
+    edges = feature.canny(image, sigma, low_threshold= low_thresh, high_threshold= hi_thresh)
+
+    low = rad_range[0]
+    hi = rad_range[1]
+    step = rad_range[2]
+
+    # Range of possible radii
+    hough_radii = np.arange(low, hi, step)
+    hough_res = transform.hough_circle(edges, hough_radii)
+
+    # Selects the most prominent circles, up to num_rings
+    accums, cx, cy, radii = transform.hough_circle_peaks(hough_res, hough_radii,
+                                               total_num_peaks=num_rings)
+
+    # Draw them
+    fig, ax = plt.subplots(ncols=1, nrows=1, figsize=(10, 4))
+    Z = color.gray2rgb(image)
+    for center_y, center_x, radius in zip(cy, cx, radii):
```

Comment:
The function _returns_ once center position but internally uses separate centers here to draw the figure. Have you considered using `center_pt` here instead? The figure might look a little worse but it will be a more accurate representation of what information is flowing downstream to the next step in the analysis, which won't have access to `cx` and `cy`, only `center_pt` and `radii`.

Reply:
Right, I thought about this too and decided to at least use the averaged center for the plotting to give the user a better visual gauge of whether the center is where they think it should be. I was thinking of including the center coords of all the rings as some return value. I could do that, but for the plotting do you think it would be helpful to plot all the centers or leave it as the averaged one?

Reply:
I just made the changes you suggested and added the centers to be part of what's returned!

Reply:
nice idea to return average center and the center for each ring

Reply:
Seems good. It might be worth considering shortening the return value to `centers, radii` because it is easy enough for the caller to then compute the average center---and, say, the standard deviation or any other stats---if they want them.

The case for leaving it `center_pt` in, as it is now, would be as a "convenience" if we strongly expect users to be interested in the average center specifically as the primary result of interest.